lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -71,16 +70,30 @@ export class BrowserCacheManager extends CacheManager {
     }
 
     /**
-     *
+     * Returns a window storage class implementing the IWindowStorage interface that corresponds to the configured temporaryCacheLocation.
+     * @param temporaryCacheLocation
      * @param cacheLocation
      */
-    protected setupTemporaryCacheStorage(cacheLocation: BrowserCacheLocation | string): IWindowStorage<string> {
-        switch (cacheLocation) {
+    protected setupTemporaryCacheStorage(temporaryCacheLocation: BrowserCacheLocation | string, cacheLocation: BrowserCacheLocation | string): IWindowStorage<string> {
+        // Create a new variable to avoid modifying the function parameter
+        let tempCacheLocation = temporaryCacheLocation;
+        if (!tempCacheLocation) {
+            /*
+             * When users do not explicitly choose their own temporaryCacheLocation and cacheLocation is set to LocalStorage
+             * Temporary cache items will always be stored in session storage to mitigate problems caused by multiple tabs
+             */
+            if (cacheLocation === BrowserCacheLocation.LocalStorage) {   
+                tempCacheLocation = BrowserCacheLocation.SessionStorage;
+            } else {
+                tempCacheLocation = cacheLocation;
+            }
+        }
+    
+        switch (tempCacheLocation) {
             case BrowserCacheLocation.LocalStorage:
             case BrowserCacheLocation.SessionStorage:
                 try {
-                    // Temporary cache items will always be stored in session storage to mitigate problems caused by multiple tabs
-                    return new BrowserStorage(BrowserCacheLocation.SessionStorage);
+                    return new BrowserStorage(tempCacheLocation);
```

Comment:
I think all of this can be simplified with just the following change: ```suggestion
                    return new BrowserStorage(temporaryCacheLocation || BrowserCacheLocation.SessionStorage;);
```

Reply:
Agreed, the first implementation was more verbose